src/luarocks/cmd/install.lua

```diff
@@ -29,22 +29,40 @@ or a filename of a locally available rock.
                     in the configuration file.
 
 --only-deps         Installs only the dependencies of the rock.
+
 --no-doc            Installs the rock without its documentation.
+
+--verify            Verify signature of the rock being installed.
+                    If rock is being downloaded, LuaRocks will attempt
+                    to download the signature as well. If the rock is
+                    local, the signature file should be in the same
+                    directory.
+                    You need the signer’s public key in your local
```

Comment:
This shouldn't be required IMO: luarocks should be verifying that the package was signed by the luarocks.org user that uploaded for it. Otherwise I could sign someone else's package

Reply:
> luarocks should be verifying that the package was signed by the luarocks.org user that uploaded for it

You mean (1) luarocks.org should verify at upload time? (Assuming the user has a way to upload their key to luarocks.org beforehand)

Or do you mean (2) that luarocks CLI should download the user key from luarocks.org for verifying? (Assuming the user has a way to upload their key to luarocks.org beforehand and the CLI has a way to download it)

AFAICT, any of these two options above would protect a user in case an API key was stolen, but not really in case luarocks.org was compromised (in which case the user public keys could be tampered too), in which case one would need --verify to check against a local key that was obtained from a third-party trusted source.

For the --sign and --verify features in this PR I basically copied the featureset from git (that sentence "You need the signer’s public key in your local..." was literally lifted from the git documentation).

Option (1) could be added to luarocks.org with no changes to this PR. So my reasoning is that this would be sufficient for a v1 of signing functionality.

Reply:
> You mean (1) luarocks.org should verify at upload time? 

It probably should; but that would be separate

> Assuming the user has a way to upload their key to luarocks.org beforehand

They should! otherwise how will we know who is allowed to sign which packages?
I would make this a dependency of the `--verify` half of the PR. But not of `--sign` which can be done first.

> in case luarocks.org was compromised (in which case the user public keys could be tampered too

In such a compromise, you could ask users to mark the gpg keys in their account as correct/valid. And then be suspicious at any signatures by unknown/untrusted keys.